src/formats.c

```diff
@@ -607,9 +607,9 @@ static char* is_key_right(struct fmt_main *format, int index,
 
 	if (i == -1) {
 		if (options.verbosity > VERB_LEGACY)
-			snprintf(err_buf, sizeof(err_buf), "cmp_one(%d) %s", match, ciphertext);
+			snprintf(err_buf, sizeof(err_buf), "crypt_all(%d) %s", index + 1, ciphertext);
 		else
-			sprintf(err_buf, "cmp_one(%d)", match);
+			sprintf(err_buf, "crypt_all(%d)", index + 1);
 		return err_buf;
 	}
```

Comment:
@solardiz If you don't have time to review the rest, perhaps you can just agree or disagree on this change of wording. I always got confused by `FAILED (cmp_one(0))` actually not pointing to a problem with `cmp_one` but with `crypt_all` returning zero after failing internal compare or early rejection.

Reply:
While we're at it, maybe when `match == count && count == index + 1` we can just test `cmp_one` for `index` once, omitting the `for` loop? Or do we have any formats that can reorder the outputs while keeping their count? Not material for this PR, though - just bringing up the idea here.

Reply:
When match was 0, `i == -1` here and we never even call `cmp_one` yet we state `FAILED (cmp_one(0))`. What good is that?

We could add some text that `crypt_all` returned zero, but there's no other way `crypt_all()` can fail so it's kind of given.

Reply:
How about reverting it and then insert a new test after `cmp_all` right before the for loop, that catches `match == 0` and says something like `FAILED (crypt_all(n) zero return)`?

Reply:
I'm OK with the separate check for `match == 0` like you have implemented now. Thanks.